### PR TITLE
Add login prompt dialog to program cards page

### PR DESCRIPTION
**Pull request by greg-hyde**
### Description

Adds the login prompt modal dialog to the program cards page. When the user is a guest, pop up a modal encouraging them to sign in and redirecting them to the sign in page or . As of right now, we don't yet have the logic to only display it once (checking local storage).

4UZPnS5BAEhdz2W

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### User visible changes

- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order

#### New Features

- [x] Conditioned new functionality on a [FeatureFlag](https://github.com/civiform/civiform/wiki/Feature-Flags)
- [x] Wrote browser tests with the feature flag off and on, etc

### CODE REVIEW

**Comment by greg-hyde** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/civiform/civiform/7540?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#7540** <a href="https://app.graphite.dev/github/pr/civiform/civiform/7540?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7478** <a href="https://app.graphite.dev/github/pr/civiform/civiform/7478?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#7492](https://github.com/civiform/civiform/pull/7492) <a href="https://app.graphite.dev/github/pr/civiform/civiform/7492?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @greg-hyde and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: server/app/assets/stylesheets/styles.css [hunk 1]
```diff
@@ -43,6 +43,11 @@
     color: #005ea2;
   }
 
+  .usa-button.usa-modal__close {
+    background-color: transparent;
+    color: #71767a;
```
Comment (gwendolyngoetz) [inline server/app/assets/stylesheets/styles.css:48]: You don't need to do it in this PR. When you have a change can you move all the hex colors into sass variables.

File: server/app/assets/stylesheets/styles.css [hunk 2]
```diff
@@ -43,6 +43,11 @@
     color: #005ea2;
   }
 
+  .usa-button.usa-modal__close {
+    background-color: transparent;
```
[the remark this answers is not in the document]
Comment (greg-hyde) [reply server/app/assets/stylesheets/styles.css:47]: unfortunately yes. If you look above, we apply a style to .usa-button:not(.usa-button--outline) that was added for J2HTML and we basically have to reset it back to the default style. These button styles are all temporary (see the TODO there) but we have to recreate it while we have a mix of uswds buttons and non-uswds buttons

File: server/app/assets/stylesheets/styles.css (hunk 1 above)
Comment (greg-hyde) [reply server/app/assets/stylesheets/styles.css:48]: will do! filed https://github.com/civiform/civiform/issues/7592

File: server/app/assets/stylesheets/styles.css (hunk 2 above)
[the remark this answers is not in the document]
Comment (gwendolyngoetz) [reply server/app/assets/stylesheets/styles.css:47]: Ahh

**Review by gwendolyngoetz — APPROVED** [review]

**Pull request merged**